core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2342,6 +2342,18 @@ private[spark] object Utils extends Logging {
    * Return the initial number of executors for dynamic allocation.
    */
   def getDynamicAllocationInitialExecutors(conf: SparkConf): Int = {
+    if (conf.get(DYN_ALLOCATION_INITIAL_EXECUTORS) < conf.get(DYN_ALLOCATION_MIN_EXECUTORS)) {
+      logWarning(s"${DYN_ALLOCATION_INITIAL_EXECUTORS.key} less than " +
+        s"${DYN_ALLOCATION_MIN_EXECUTORS.key} is invalid, will use " +
+          s"${DYN_ALLOCATION_MIN_EXECUTORS.key} to override.")
```

Comment:
I think my original meaning is `DYN_ALLOCATION_MIN_EXECUTORS` will override `DYN_ALLOCATION_INITIAL_EXECUTORS` if the former is larger. Not related to num executor instance, num executor instance can still be larger than `DYN_ALLOCATION_MIN_EXECUTORS`.